### DISCUSSION TITLE
Get frequency data from a Media Player

### DISCUSSION THREAD

**Discussion by BrandMChaos** [naudio/NAudio#1116] [Ideas]
Is it possible to use this library to use something like Fast Fourier Transform (in this case specifically Discrete-Time Fourier Transform) on output audio ?

Right now I'm trying to use this data to turn it into instructions that will have motors move in sync to a song for example

![image](https://github.com/naudio/NAudio/assets/68779123/eea12ae5-ef5b-4661-b8b9-e34e6d64b140)

**Comment by markheath** [comment]
NAudio does provide an FFT implementation (see FastFourierTransform) and in NAudio.Extras there is a class called SampleAggregator which performs FFTs on the audio as it passes through an ISampleProvider which you could adapt for your own needs.
